backend/python/app/routes/entity_routes.py

```diff
@@ -0,0 +1,75 @@
+from flask import Blueprint, request
+from flask import jsonify
+
+from ..resources.entity_dto import EntityDTO
+from ..services import entity_service
+
+# defines a shared URL prefix for all routes
+blueprint = Blueprint("entity", __name__, url_prefix="/entities")
+
+# defines GET endpoint for retrieving all entities
+@blueprint.route("/", methods=["GET"], strict_slashes=False)
+def get_entities():
+    result = entity_service.get_entities()
+    return jsonify(result), 200
+
+
+# defines GET endpoint for retrieving a single entity based on a provided id
+@blueprint.route("/<int:id>", methods=["GET"], strict_slashes=False)
+def get_entity(id):
+    try:
+        result = entity_service.get_entity(id)
+    except:
+        error = {"error": "entity not found"}
+        # HTTP status code 404 means Not Found
+        return jsonify(error), 404
```

Comment:
Going to deviate a bit from the bootcamp, I think it would make more sense to treat all service layer errors as 500 (server) errors and report the error message exactly as given from the service layer